app/lib/actions/katello/bulk_agent_action.rb

```diff
@@ -3,19 +3,32 @@ module Katello
     class BulkAgentAction < Actions::BulkAction
       def plan(agent_action, hosts, args)
         host_ids = hosts.map(&:id)
-        dispatch_args = {
-          content: args[:content]
-        }
-        histories = ::Katello::Agent::Dispatcher.dispatch(agent_action.agent_message, host_ids, dispatch_args)
+
+        histories = ::Katello::Agent::Dispatcher.create_histories(
+          host_ids: host_ids
+        )
 
         grouped_histories = {}
         histories.each { |h| grouped_histories[h.host_id] = h.id }
         options = {
           dispatch_histories: grouped_histories,
-          content: args[:content]
+          type: agent_action.agent_message,
+          content: args[:content],
+          bulk: true
         }
         super(agent_action, hosts, options)
       end
+
+      def spawn_plans
```

Comment:
who calls this ?

Reply:
Dynflow! https://github.com/Dynflow/dynflow/blob/d7bde98a9ac370c64e0ee2532758481d4956bb9c/lib/dynflow/action/with_bulk_sub_plans.rb#L18